emma/OntoEmma.py

```diff
@@ -184,33 +182,49 @@ def load_alignment(self, gold_path):
                 )
 
     def train(
-        self, model_type: str, model_path: str, config_file: str
+        self, model_type: str, model_path: str, config_file: str, cuda_device: int = -1
```

Comment:
I did this initially because it was easier to change cuda devices quickly for experiments, but I think it makes more sense to have the cuda device specified in the config file and nowhere else. I will change this along with some of the other refactoring below.